src/server/api/api.ts

```diff
@@ -53,18 +53,18 @@ export const websocketHandler: WebsocketRequestHandler = (ws, req) => {
                 throw new Error("Game must be started first.");
             }
 
-            chess.move({ from: message.from, to: message.to });
+            chess.makeMove(message.from, message.to);
 
-            if (chess.isGameOver()) {
+            if (chess.getGameFinishedReason() != undefined) {
                 // Game is naturally finished; we're done
                 return;
             }
 
             // Absolutely unhinged api design
-            const val = Object.entries(aiMove(chess.fen(), difficulty))[0];
+            const val = Object.entries(aiMove(chess.fen, difficulty))[0];
             const from = val[0].toLowerCase();
             const to = (val[1] as string).toLowerCase();
-            chess.move({ from, to });
+            chess.makeMove(from, to);
```

Comment:
Chess should also have a `makeAiMove(difficulty)` method. We should also refactor `difficulty` into an explicit string enum. Ask me for help updating the UI to use the enum once you create it.